chapter06/spg/exercise1.pl

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Storable qw( retrieve nstore );
+
+my $output = "stored_data.dat";
+
+my $data = {};
+
+if ( -s $output ) {
+  $data = retrieve( $output );
+}
+
+while ( <> ) {
+  my ( $source, $dest, $transfer ) = split;
+  $data->{$source}->{$dest} += $transfer;
+}
+
+nstore( $data, $output );
```

Comment:
Hadn't thought about what would happen if the output file was read only or broken. Looks like Storable throws a die error.